include/envoy/api/io_error.h

```diff
@@ -25,6 +25,14 @@ class IoError {
     InProgress,
     // Permission denied.
     Permission,
+    // Message too big to send.
+    MessageTooBig,
+    // Kernel interrupt.
+    Interrupt,
```

Comment:
What does this mean / when does this happen?

Reply:
This maps to EINTR. recvmsg/sendmsg can be interrupted by signal occasionally. This happens a lot on blocking socket, but I'm not 100% sure it won't happen on non-blocking socket. If it does happen on non-blocking socket, it shouldn't be treated like one of any other errors, because nothing actually goes wrong and we should try again.